PlayerSkeleton.java

```diff
@@ -1,77 +1,83 @@
+     * @param s (current state), legalMoves (possible moves for current piece)
+     * @return move with best (highest) heuristic score
+     **/
+    private int tryPossibleMoves(State s, int[][] legalMoves, double[] weights) {
+        //legalMoves = all possible moves of the current piece.
+        //Step 1. Apply each move (action) to get the change field (board configuration).
+        //Step 2. Apply Heuristic.evaluate(s) to get the heuristic score from the move.
+        //Step 3. Get the best move.
+        double bestScore = Double.NEGATIVE_INFINITY;
+        int bestMove = 0;
 
-		for(int i=0; i<legalMoves.length; i++) {
-			TempState ts = new TempState(s);
-			int prevCleared = s.getRowsCleared();
-			//Step 1
-			int orient = legalMoves[i][0];
-			int slot = legalMoves[i][1];
-			boolean possible = ts.makeMove(orient, slot);
-			if(!possible) {
-				continue;
-			} 
-			//Step 2
+        for(int i=0; i<legalMoves.length; i++) {
+            TempState ts = new TempState(s);
+            int prevCleared = s.getRowsCleared();
+            //Step 1
+            int orient = legalMoves[i][0];
+            int slot = legalMoves[i][1];
+            boolean possible = ts.makeMove(orient, slot);
+            if(!possible) {
+                continue;
+            }
+            //Step 2
             ts.setPrevCleared(prevCleared);
-			ts.setOrientAndSlot(orient, slot);
-			double score = HeuristicNew.evaluate(ts, weights);
+            ts.setOrientAndSlot(orient, slot);
+            HeuristicNew heuristic = new HeuristicNew(ts, weights);
+            double score = heuristic.evaluate();
 
-			//Step 3
-			if(score > bestScore) {
-				bestScore = score;
-				bestMove = i;
-			}
-		}
-		return bestMove;
-	}
-	
-/*	public static void main(String[] args) {
-		State s = new State();
-		new TFrame(s);
-		PlayerSkeleton p = new PlayerSkeleton();
-		double[] weights = {-0.4944712922445228, 0.14814799954876456, -0.9250354407343329, -1.210228068000284, -0.6363751749245313, -0.3071493143829803,
-				0.4062421734216026, 0.4053785630482645, 0.7203562102658518, 0.6458317403226688, 0.33541924448498417,
-				0.156339060819642, 1.2098194382476808, -0.12143020086913736, 1.5266470763709517, 0.5760539680303807,
-				0.11214199088469634, -0.2072277305271486, 1.2884998585410108, 0.9616177885473564, 0.13438465406024958, 1.1912072595739716};
-		while(!s.hasLost()) {
-			s.makeMove(p.pickMove(s, s.legalMoves(), weights));
-			s.draw();
-			s.drawNext(0,0);
-			try {
-				Thread.sleep(0);
-			} catch (InterruptedException e) {
-				e.printStackTrace();
-			}
-		}
-		System.out.println("You have completed "+s.getRowsCleared()+" rows.");
-	}*/
+            //Step 3
+            if(score > bestScore) {
+                bestScore = score;
+                bestMove = i;
+            }
+        }
+        if (trackMoveSelection) System.out.println("Picked best move for " + Arrays.toString(weights) + " with score of [" + bestScore + "]");
```

Comment:
Moves the best move selection statement to here and added a switch trackMoveSelection.

HeuristicsNew works as intended in its new OOP implementation